src/Asset/AssetBundle.php

```diff
@@ -67,113 +68,117 @@ class AssetBundle
+
     /**
-     * @var array list of CSS files that this bundle contains. Each CSS file can be specified
-     *            in one of the three formats as explained in [[js]].
-     *
-     * Note that only a forward slash "/" should be used as directory separator.
-     */
-    public $css = [];
-    /**
-     * @var array the options that will be passed to [[View::registerJsFile()]]
-     *            when registering the JS files in this bundle.
+     * @var array the options that will be passed to {@see View::registerJsFile()} when registering the JS files in this
+     *            bundle.
      */
     public $jsOptions = [];
+
     /**
-     * @var array the options that will be passed to [[View::registerCssFile()]]
-     *            when registering the CSS files in this bundle.
-     */
-    public $cssOptions = [];
-    /**
-     * @var array the options to be passed to [[AssetManager::publish()]] when the asset bundle
-     *            is being published. This property is used only when [[sourcePath]] is set.
+     * @var array the options to be passed to {@see AssetManager::publish()} when the asset bundle  is being published.
+     *            This property is used only when {@see sourcePath} is set.
      */
     public $publishOptions = [];
 
     /**
-     * Registers this asset bundle with a view.
+     * @var string the directory that contains the source asset files for this asset bundle. A source asset file is a
+     *             file that is part of your source code repository of your Web application.
+     *
+     * You must set this property if the directory containing the source asset files is not Web accessible. By setting
+     * this property, [[AssetManager]] will publish the source asset files to a Web-accessible directory automatically
+     * when the asset bundle is registered on a page.
      *
-     * @param View $template the view to be registered with
+     * If you do not set this property, it means the source asset files are located under {@see basePath}.
      *
-     * @return static the registered asset bundle instance
+     * You can use either a directory or an alias of the directory.
+     *
+     * {@see publishOptions}
      */
-    public static function register($template)
-    {
-        return $template->registerAssetBundle(get_called_class());
-    }
+    public $sourcePath;
 
     /**
-     * Initializes the bundle.
-     * If you override this method, make sure you call the parent implementation in the last.
+     * Publishes the asset bundle if its source code is not under Web-accessible directory.
+     *
+     * It will also try to convert non-CSS or JS files (e.g. LESS, Sass) into the corresponding CSS or JS files using
+     * {@see AssetManager::converter|asset converter}.
+     *
+     * @param AssetManager $am the asset manager to perform the asset publishing
+     * @return void
      */
-    public function init(): void
+    public function publish(AssetManager $am): void
```

Comment:
Is it the job of the bundle to publish itself?

Reply:
It is still valid AssetBundle publishes itself with the help of asset manager. Maybe asset manager should do it completely?